addons/stock/product.py

```diff
@@ -392,153 +373,147 @@ def _compute_nbr_reordering_rules(self, cr, uid, ids, field_names=None, arg=None
+    property_stock_inventory = fields.Many2one(
+        company_dependent=True,
+        comodel_name='stock.location',
+        string="Inventory Location",
+        domain=[('usage', 'like', 'inventory')],
+        help="This stock location will be used, instead of the default one, as the source location for stock moves generated when you do an inventory.")
+    sale_delay = fields.Float('Customer Lead Time', help="The average delay in days between the confirmation of the customer order and the delivery of the finished products. It's the time you promise to your customers.", default=7)
+    tracking = fields.Selection(selection=[('serial', 'By Unique Serial Number'), ('lot', 'By Lots'), ('none', 'No Tracking')], string="Tracking", required=True, default='none')
+    description_picking = fields.Text('Description on Picking', translate=True)
+    # sum of product variant qty
+    # 'reception_count': fields.function(_product_available, multi='qty_available',
+    #     fnct_search=_search_product_quantity, type='float', string='Quantity On Hand'),
+    # 'delivery_count': fields.function(_product_available, multi='qty_available',
+    #     fnct_search=_search_product_quantity, type='float', string='Quantity On Hand'),
+    qty_available = fields.Float(compute="_product_available",
+        search="_search_product_quantity", string='Quantity On Hand')
+    virtual_available = fields.Float(compute="_product_available",
+        search="_search_product_quantity", string='Forecasted Quantity')
+    incoming_qty = fields.Float(compute="_product_available",
+        search="_search_product_quantity", string='Incoming')
+    outgoing_qty = fields.Float(compute="_product_available",
+        search="_search_product_quantity", string='Outgoing')
+    location_id = fields.Many2one(string='Location', comodel_name='stock.location')
+    warehouse_id = fields.Many2one(string='Warehouse', comodel_name='stock.warehouse')
+    route_ids = fields.Many2many('stock.location.route', 'stock_route_product', 'product_id', 'route_id', 'Routes', domain="[('product_selectable', '=', True)]",
+                                help="Depending on the modules installed, this will allow you to define the route of the product: whether it will be bought, manufactured, MTO/MTS,...")
+    nbr_reordering_rules = fields.Integer(compute="_compute_nbr_reordering_rules", string='Reordering Rules')
+    reordering_min_qty = fields.Float(compute="_compute_nbr_reordering_rules")
+    reordering_max_qty = fields.Float(compute="_compute_nbr_reordering_rules")
+    route_from_categ_ids = fields.Many2many(related='categ_id.total_route_ids', comodel_name="stock.location.route", string="Category Routes")
+
+    @api.multi
+    def action_view_routes(self):
+        route_obj = self.Env["stock.location.route"]
+        act_obj = self.env['ir.actions.act_window']
+        mod_obj = self.env['ir.model.data']
```

Comment:
Give Proper Object variable name : 
```
StockLocationRoute = self.Env["stock.location.route"]
IrAction = self.env['ir.actions.act_window']
IrModel = self.env['ir.model.data']
```
For more information follow [Guideline](http://odoo-new-api-guide-line.readthedocs.org/en/latest/)